designer/server/src/test/scala/pl/touk/nussknacker/ui/process/ProcessStateDefinitionServiceSpec.scala

```diff
@@ -72,51 +61,90 @@ class ProcessStateDefinitionServiceSpec extends AnyFunSuite with Matchers {
+    val fraudProcessStateDefinitionManager = createStateDefinitionManager(Map("COMMON" -> "Common", "CUSTOM_FRAUD" -> "Fraud"))
 
-    val providerWithInvalidStateDefinitions = processingTypeDataProvider(
+    val definitions = testStateDefinitions(
+      CommonUser("user", "user", Map(Category1 -> Set(Permission.Read))),
       streamingProcessStateDefinitionManager,
       fraudProcessStateDefinitionManager
     )
 
+    definitions should have size 2
+
+    val expectedCommon = streamingProcessStateDefinitionManager.stateDefinitions("COMMON")
+    definitions should contain(UIStateDefinition(
+      name = "COMMON",
+      displayableName = expectedCommon.displayableName,
+      icon = expectedCommon.icon,
+      tooltip = expectedCommon.tooltip,
+      categories = List(Category1)
+    ))
+
+    val expectedCustomStreaming = streamingProcessStateDefinitionManager.stateDefinitions("CUSTOM_STREAMING")
+    definitions should contain(UIStateDefinition(
+      name = "CUSTOM_STREAMING",
+      displayableName = expectedCustomStreaming.displayableName,
+      icon = expectedCustomStreaming.icon,
+      tooltip = expectedCustomStreaming.tooltip,
+      categories = List(Category1)
+    ))
+  }
+
+  test("should raise exception when definitions with the same name are NOT unique") {
+    val streamingProcessStateDefinitionManager = createStateDefinitionManager(Map("COMMON" -> "Non unique name", "CUSTOM_STREAMING" -> "Streaming"))
+    val fraudProcessStateDefinitionManager = createStateDefinitionManager(Map("COMMON" -> "Common", "CUSTOM_FRAUD" -> "Fraud"))
+
     intercept[IllegalStateException]{
-      ProcessStateDefinitionService.validate(providerWithInvalidStateDefinitions.all)
+      testStateDefinitions(
+        AdminUser("admin", "admin"),
+        streamingProcessStateDefinitionManager,
+        fraudProcessStateDefinitionManager
+      )
     }.getMessage should include("State definitions are not unique")
   }
 
+}
 
-  private class StreamingProcessStateDefinitionManager(displayableName: String = "Common",
-                                                       icon: Option[URI] = None,
-                                                       tooltip: Option[String] = None,
-                                                       description: Option[String] = Some("This definition is common for all processing types")
-                                                      ) extends OverridingProcessStateDefinitionManager(
-    customStateDefinitions = Map(
-      "COMMON" -> StateDefinitionDetails(
-        displayableName = displayableName, icon = icon, tooltip = tooltip, description = description
-      ),
-      "CUSTOM_STREAMING" -> StateDefinitionDetails(
-        displayableName = "Streaming", icon = None, tooltip = None, description = Some("This definition is specific for stremaing")
-      ),
-    )
-  )
+object ProcessStateDefinitionServiceSpec {
```

Comment:
I prefer to move these private fields and methods just to class ProcessStateDefinitionServiceSpec -> companion object is useless. But I left the decision to you ;)

Reply:
I've moved methods from companion object and made them private.